app/cdap/components/DirectiveInput/Components/InputPanel/index.tsx

```diff
@@ -0,0 +1,248 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Box, Typography } from '@material-ui/core';
+import MyDataPrepApi from 'api/dataprep';
+import { defaultFuseOptions } from 'components/DirectiveInput/constants';
+import { IDirectiveUsage, IObject } from 'components/DirectiveInput/types';
+import ee from 'event-emitter';
+import Fuse from 'fuse.js';
+import reverse from 'lodash/reverse';
+import Mousetrap from 'mousetrap';
+import React, { useEffect, useState } from 'react';
+import globalEvents from 'services/global-events';
+import NamespaceStore from 'services/NamespaceStore';
+import styled from 'styled-components';
+import uuidV4 from 'uuid/v4';
+import { grey } from '@material-ui/core/colors';
+import { IHeaderNamesList } from 'components/GridTable/types';
+
+interface IInputPanelProps {
+  setDirectivesList: React.Dispatch<React.SetStateAction<[]>>;
+  selectedDirective: boolean;
+  columnNamesList: IHeaderNamesList[];
+  onSearchItemClick: (value: string) => void;
+  getDirectiveSyntax: (results: IDirectiveUsage[], value: boolean) => void;
+  onColumnSelection: (value: boolean) => void;
+  inputBoxValue: string;
+}
+
+const SimpleWrapper = styled(Box)`
+  display: block;
+`;
+
+const SmallLabel = styled(Typography)`
+  font-style: normal;
+  font-weight: 400;
+  font-size: 14px;
+  letter-spacing: 0.15;
+  color: ${grey[700]};
+`;
+
+const LargeLabel = styled(SmallLabel)`
+  font-weight: 600;
+  font-size: 16px;
+`;
+
+const ResultRow = styled(Box)`
+  padding: 10px;
+  border-bottom: 1px solid ${grey[300]};
+  background-color: #ffffff;
+  &:hover {
+    background: #eff0f2;
+    cursor: pointer;
+  }
+`;
+
+const ActiveResultRow = styled(ResultRow)`
+  background-color: #eff0f2;
+`;
+
+export default function({
+  setDirectivesList,
+  selectedDirective,
+  columnNamesList,
+  onSearchItemClick,
+  getDirectiveSyntax,
+  onColumnSelection,
+  inputBoxValue,
+}: IInputPanelProps) {
+  const [searchResults, setSearchResults] = useState<IDirectiveUsage[]>([]);
+  const [inputText, setInputText] = useState<string>('');
+  const [selectedIndex, setSelectedIndex] = useState<number | null>(null);
+  const eventEmitter = ee(ee);
+  const [fuse, setFuse] = useState(new Fuse([], { ...defaultFuseOptions }));
+
+  const getUsage = () => {
+    if (!selectedDirective) {
+      MyDataPrepApi.getUsage({ context: NamespaceStore.getState().selectedNamespace }).subscribe(
+        (res) => {
+          setDirectivesList(res.values);
+          setFuse(new Fuse(res.values, { ...defaultFuseOptions }));
+        }
+      );
+    } else {
+      setFuse(new Fuse(columnNamesList, { ...defaultFuseOptions, keys: ['label'] }));
+    }
+  };
+
+  useEffect(() => {
+    getUsage();
+  }, [selectedDirective]);
+
+  useEffect(() => {
+    eventEmitter.on(globalEvents.DIRECTIVEUPLOAD, getUsage);
```

Comment:
this file needs a lot more comments explaining whats going on and why

Reply:
added comments